src/test/java/com/exasol/glue/ittests/GlueLocalValidationIT.java

```diff
@@ -68,65 +73,122 @@ void beforeEach() {
+        final DynamicFrame dyf = getGlueSource(table.getFullyQualifiedName()).getDynamicFrame();
         final Dataset<Integer> ds = getDynamicFrameAsDataset(dyf, Encoders.INT());
         assertAll(() -> assertThat(dyf.count(), equalTo(6L)),
                 () -> assertThat(ds.collectAsList(), containsInAnyOrder(1, 2, 3, 4, 5, 6)));
     }
 
     @Test
     void testApplyMapping() {
-        this.options.put("table", table.getFullyQualifiedName());
         final Seq<Product> mappings = JavaConverters
                 .asScalaBuffer(Arrays.asList(Tuple4.apply("c1", "int", "c1", "string")));
-        final DynamicFrame dyf = getGlueSource().getDynamicFrame().applyMapping(mappings, true, "", null, 0, 0);
+        final DynamicFrame dyf = getGlueSource(table.getFullyQualifiedName()) //
+                .getDynamicFrame() //
+                .applyMapping(mappings, true, "", null, 0, 0);
         final Dataset<String> ds = getDynamicFrameAsDataset(dyf, Encoders.STRING());
         assertThat(ds.collectAsList(), containsInAnyOrder("1", "2", "3", "4", "5", "6"));
     }
 
     @Test
-    void testReadWrite() {
-        this.options.put("table", table.getFullyQualifiedName());
-        final DynamicFrame dyf = getGlueSource().getDynamicFrame();
+    void testReadExasolWriteS3() {
+        final DynamicFrame dyf = getGlueSource(table.getFullyQualifiedName()).getDynamicFrame();
 
         // Save DynamicFrame
         final String writePath = "/tmp/output/readwritetest/" + UUID.randomUUID();
-        final JsonOptions writeOptions = new JsonOptions("{\"separator\":\",\", \"withHeader\": true}");
         glueContext //
-                .getSinkWithFormat("s3", new JsonOptions("{\"path\":\"" + writePath + "\"}"), "", "csv", writeOptions) //
+                .getSinkWithFormat("s3", new JsonOptions("{\"path\":\"" + writePath + "\"}"), "", "csv",
+                        getCSVWriteOptions()) //
                 .writeDynamicFrame(dyf, CallSite.apply("", ""));
 
         // Read Saved DynamicFrame
         final DynamicFrame readDyf = glueContext //
-                .getSource("s3", new JsonOptions("{\"paths\":[\"" + writePath + "\"]}"), "", "") //
-                .withFormat("csv", writeOptions) //
+                .getSource("s3", getCSVPaths(writePath), "", "") //
+                .withFormat("csv", getCSVWriteOptions()) //
                 .getDynamicFrame();
 
         assertAll(() -> assertThat(readDyf.count(), equalTo(6L)),
                 () -> assertThat(getDynamicFrameAsDataset(readDyf, Encoders.STRING()).collectAsList(),
                         containsInAnyOrder("1", "2", "3", "4", "5", "6")));
     }
 
-    private DataSource getGlueSource() {
-        return glueContext.getSource("custom.spark", getJsonOptions(), "customSource", "");
+    @Test
+    void testReadS3WriteExasol() throws SQLException {
+        final String writePath = new File("src/test/resources/data/").getAbsolutePath();
```

Comment:
External resource files make tests hard to read. Add a method that creates a temporary file with the values given as parameters instead. This way reviewers can see test preparation an assertion in the same place and the code is still compact.

Reply:
Going to address in #37